### ISSUE TITLE
Add unique generated tileset IDs

### ISSUE THREAD

**Issue by brendan-ward** [consbio/mbtileserver#94]
As discussed in #92 

Add option `--generate-ids`, default is `false`.

If present, each tileset will get a URL-encoded SHA1 hash that represents its path and filename.  This has 2 implications:
1. all services are effectively flat, no deeper than `/services/<sha1>/`
2. this allows duplicate paths and filenames without causing URL collisions

Example hashing:
```
hash := sha1.New()
hash.Write([]byte(filename))
id = base64.URLEncoding.EncodeToString(hash.Sum(nil))
```

To go along with this, we may want to add the source path (relative to tile root directory) for each tileset to the `services/` and tileJSON endpoints.

**Comment by asmyasnikov** [comment]
Brief code will be follow:
```
id = base64.URLEncoding.EncodeToString(sha1.Sum([]byte(filename)))
```

**Comment by brendan-ward** [comment]
Resolved by #100 , included in forthcoming 0.7 release
